R/client_state.R

```diff
@@ -0,0 +1,133 @@
+client_get_state <- new_generic(
+  "client_get_state",
+  "client",
+  function(client) {
+    result <- S7::S7_dispatch()
+  }
+)
+
+client_set_state <- new_generic(
+  "client_get_state",
+  "client",
+  function(client, state) {
+    S7::S7_dispatch()
+  }
+)
+
+client_set_ui <- new_generic(
+  "client_set_ui",
+  "client",
+  function(client, ..., id) {
+    S7::S7_dispatch()
+  }
+)
+
+is_server_bookmarkstore <- function() {
+  shiny::getShinyOption("bookmarkStore", "server") == "server"
+}
+is_url_bookmarkstore <- function() {
+  shiny::getShinyOption("bookmarkStore", "url") == "url"
+}
+
+method(client_get_state, S7::new_S3_class(c("Chat", "R6"))) <-
```

Comment:
Interesting that `ellmer::Chat` is a `R6::R6Class()` , but this is `new_S3_class()`? I have know idea how/if this is supposed to work, but I'm willing to take your word for it :)

Reply:
Garrick pointed me to it. Because the `Chat` class is a "class" on the object, then standard s3 dispatch will work. 

Also, `ellmer:::Chat` is not exported as of today: https://github.com/tidyverse/ellmer/blob/2c8364f5efddca492334a2ea9b68ed70cea06f29/NAMESPACE